### PR TITLE
ComposePar Retiming

### PR DESCRIPTION
**Pull request by akeley98**
Add ComposePar retiming passes (match latencies on all paths through a ComposePar).

Add ComposePar retiming tests to TestCompose.hs.

Update simulator readme to new op syntax.

Change function names as requested.

Fix missing images preventing Gaussian and MipMap examples from running.

### CODE REVIEW

File: src/Core/Aetherling/Passes/Timing.hs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+-- | Given an AST, distribute all LogicalUtil ops to leaf ops, so that
+-- each leaf op in the output AST is wrapped (if needed) with a
+-- LogicalUtil whose ratio is the product of the LogicalUtil ratios on
+-- the path from the leaf to the root (except at ReadyValid
+-- boundaries, the ratio is reset to 1). This allows us to see the
+-- real speed of an op without inspecting the whole AST. Named by
+-- analogy to mathematical distribution A(x+y) -> Ax+Ay. (A is the
+-- metaphorical LogicalUtil, x and y the ops).
+--
+-- Note: This will not remove Delay ops, but their delay amount will
```
Comment (David-Durst) [inline src/Core/Aetherling/Passes/Timing.hs:29]: Add to this comment explaining that retimeComposePar can be run after this op to solve for the correct delay amounts. Thus, setting them all to 1 in distirbuteUtil not a problem

File: src/Core/Aetherling/Passes/Timing.hs [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+distributeUtil :: Op -> Op
+distributeUtil = distributeUtilImpl 1
+
+-- Implementation function for distributeUtil; the Ratio Int arg keeps
+-- track of the product of all LogicalUtil ratios in the path to the
+-- root (or to a ReadyValid meta-op).
+distributeUtilImpl :: (Ratio Int) -> Op -> Op
+distributeUtilImpl ratio0 (LogicalUtil ratio1 op) =
+  distributeUtilImpl (ratio0 * ratio1) op
+distributeUtilImpl ratio (ReadyValid op) =
```
Comment (David-Durst) [inline src/Core/Aetherling/Passes/Timing.hs]: please get rid of the special case for readyvalid. This is unintuitive as it makes stuff disappear.

File: src/Core/Aetherling/Passes/Timing.hs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+-- Settings for retimeComposePar. More may be added with time.
+data RetimeComposeParPolicy = RetimeComposeParPolicy {
+  rcpRetimeReadyValid_ :: Bool
+}
+
+-- Internal data type used in recursive implementation function
+-- for retimeComposePar. See function for field meanings.
+data RetimeComposeParResult = RetimeComposeParResult {
+  rcpLowDelay :: Op,
+  rcpHighDelay :: Op,
```
Comment (David-Durst) [inline src/Core/Aetherling/Passes/Timing.hs]: What are high delay and low delay? What is high latency/low latency?

**Comment by akeley98** [comment]
Replaced with pull request #26.

**Comment by David-Durst** [comment]
Note: this was not actually merged in. #26 is all the good parts of this. The parts that I do not fully accept yet, the composePar retiming file (Passes/Timing.hs) and the tests for that file in test/TestCompose.hs, have not been accepted. Thus, this pull request technically being accepted did not break the build of dev.

**Pull request merged**